source/slang/slang-lower-to-ir.cpp

```diff
@@ -6481,34 +6481,38 @@ static void lowerFrontEndEntryPointToIR(
         {
             IRParam* irParam = valueWithParams->getFirstParam();
 
-            for (auto param : params)
+            // Only decorate if this has params (ie it is a definition, not a declaration)
+            if (irParam)
```

Comment:
I think there is a utility function in the IR to test if a function is a definition vs. declaration, which might make the intent here more explicit even without the comment.